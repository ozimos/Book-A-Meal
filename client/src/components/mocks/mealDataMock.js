export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
