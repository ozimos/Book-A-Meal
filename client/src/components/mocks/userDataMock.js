export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
