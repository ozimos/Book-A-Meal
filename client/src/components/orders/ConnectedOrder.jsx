import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactModal from 'react-modal';

import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';
import SearchInput, { createFilter } from 'react-search-input';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import MealDisplayCard from '../mealCard/MealDisplayCard';
import MealRow from '../orderCart/MealRow';
import ConnectedCartContainer from '../orderCart/ConnectedCartContainer';
import MealCardContainer from '../mealCard/MealCardContainer';
import OrderContainer from '../mealCard/OrderContainer';
import MealsTable from '../mealCard/MealsTable';
import Greeting from '../greeting/Greeting';
import { menuActions, orderActions } from '../../redux/actions';
import ConnectedNav from '../nav/ConnectedNav';
import '../../../public/styles/bookameal.scss';
import '../../../public/styles/search-input.css';
import '../../../public/styles/accordion.css';

ReactModal.setAppElement(document.getElementById('root'));
class Order extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      currentOrder: [],
      currentOrderId: '',
      selectedOrderId: '',
      showOrderModal: false,
      showMealDetailModal: false,
      currentPage: 0
    };
  }

  componentDidMount() {
    const { offset = 0, limit = 5 } = this.props.pagination;
    this.props.dispatch(menuActions.getMenu());
    this.props.dispatch(orderActions.getOrdersWithMealLinks({ limit, offset }));
  }

  onFetchMealData = (state) => {
    const id = this.state.selectedOrderId || this.props.orders[0].id;
    const { page, pageSize } = state;
    const offset = pageSize * page;
    return this.props.dispatch(orderActions
      .getMealsInOrder(id, { limit: pageSize, offset }));
  }

   onFetchOrderData = (state) => {
     const { page, pageSize } = state;
     const offset = pageSize * page;
     this.props.dispatch(orderActions
       .getOrdersWithMealLinks({ limit: pageSize, offset }));
   }
   getOrderMealsTotals = id =>
     this.props.dispatch(orderActions.getOrderTotal(id))
  getOrderMeals = (id) => {
    const { offset = 0, limit = 5 } = this.props.mealsPagination;
    this.setState({ showMealDetailModal: true, selectedOrderId: id });
    this.props.dispatch(orderActions.getMealsInOrder(id, { limit, offset }));
  }

  openCartModal = () => this.setState({ showOrderModal: true })

  closeCartModal = () => this.setState({ showOrderModal: false })

  closeMealDetailModal = () => this.setState({ showMealDetailModal: false });

  notify = message => toast(message, { className: 'toaster' });

  addMealToCart = (meal) => {
    const inOrder = this.state.currentOrder.some(elem => elem.id === meal.id);
    if (!inOrder) {
      this.setState(prevState =>
        ({ currentOrder: [...prevState.currentOrder, meal] }));
      toast('Meal has been added to cart', { className: 'toaster' });
    } else {
      toast('Meal is already in cart', { className: 'toaster' });
    }
  }

  addOrderToCart = (id) => {
    const order = this.props.pendingOrders.find(elem => elem.id === id);
    this.setState({ currentOrder: order.Meals, currentOrderId: id });
    toast('Meal has been added to cart for editing', { className: 'toaster' });
  }

  removeMealFromCart = (id) => {
    if (this.state.currentOrder.length <= 1) {
      toast.error(
        `There must be at least one meal in the cart. 
        Use the clear cart button to clear cart`,
        { className: 'toaster' }
      );
    } else {
      this.setState(prevState =>
        ({ currentOrder: prevState
          .currentOrder.filter(elem => elem.id !== id) }));
    }
  }

  clearOrder = () => {
    this.setState({ currentOrder: [], currentOrderId: '' });
    this.closeCartModal();
  }

  searchUpdated = (term) => {
    this.setState({ searchTerm: term });
  }

  postOrder = () => {
    const orderIdArray = this.state.currentOrder.map(meal => meal.id);
    this.props.dispatch(orderActions.postOrder({ currentOrder: orderIdArray }));
  }

  render() {
    const KEYS_TO_FILTERS = ['id', 'title', 'description', 'price'];


    const isMenuSet = this.props.menu.length !== 0;
    const isMealSelected = this.state.currentOrder
    && this.state.currentOrder.length;
    const isTodayOrder = this.props.orders.length !== 0;
    let filteredMeals;
    if (isMenuSet) {
      filteredMeals = this.props.menu.length ? this.props.menu
        .filter(createFilter(this.state.searchTerm, KEYS_TO_FILTERS)) : [];
    }
    const { isCaterer, firstName } = this.props.user;

    return (
      <div className="contain">
        <header className="header">
          <ConnectedNav />
        </header>
        <div className="flexbox">

          <Greeting isCaterer={isCaterer} firstName={firstName} />
          <button
            className="btn title-button"
            onClick={this.openCartModal}
            disabled={!isMealSelected}
          >
            <p
              className="cart-notification"
            >
                      Cart
              <span className="badge">
                {this.state.currentOrder.length}
              </span>
            </p>
          </button>
        </div>
        <div className="row">
          <main className="col s12">
            <ToastContainer autoClose={2000} />
            <Accordion accordion>
              <AccordionItem expanded>
                <AccordionItemTitle>
                  <div className="title-element flexbox">
                    <h4 className="long_string">
                  Today&#39;s Menu
                    </h4>
                    <div
                      className="accordion__arrow u-position-relative"
                      role="presentation"
                    />
                  </div>
                </AccordionItemTitle>
                <AccordionItemBody>
                  <div className="flexbox">
                    <SearchInput
                      className="search-input input-field"
                      onChange={this.searchUpdated}
                    />

                  </div>
                  {isMenuSet ? <MealCardContainer
                    meals={filteredMeals}
                    MealCard={MealDisplayCard}
                    addToCollection={this.addMealToCart}
                    collection="cart"
                    addClass="scroll"
                  /> :
                  <div>
                  The menu for the day has not been set
                  </div>
                }
                </AccordionItemBody>
              </AccordionItem>
              <AccordionItem>
                <AccordionItemTitle>
                  <div className="title-element flexbox">
                    <h4 className="long_string">
                      Your Order History
                    </h4>
                    <div
                      className="accordion__arrow u-position-relative"
                      role="presentation"
                    />
                  </div>
                </AccordionItemTitle>
                <AccordionItemBody>
                  <p className="mr-auto">
                    {`Orders can only be edited up to 
                    ${process.env.ORDER_EDIT_MINUTES || 15}
                     minutes after order is placed`}
                  </p>
                  { isTodayOrder ? <OrderContainer
                    orders={this.props.orders}
                    loading={this.props.orderConnecting}
                    pagination={this.props.pagination}
                    addOrderToCart={this.addOrderToCart}
                    onFetchData={this.onFetchOrderData}
                    getOrderMeals={this.getOrderMeals}
                    getOrderMealsTotals={this.getOrderMealsTotals}
                    defaultPage={this.state.currentPage}
                    currentOrderId={this.state.currentOrderId}
                  /> :
                  <div>
                  You have not placed an order today
                  </div>
                }
                </AccordionItemBody>
              </AccordionItem>
            </Accordion>
          </main>

          <ReactModal
            isOpen={this.state.showOrderModal}
            contentLabel="Input Modal"
            className="modal-content"
            onRequestClose={this.closeCartModal}
            shouldCloseOnOverlayClick
          >
            <aside className="col s12" >
              {isMealSelected ? <ConnectedCartContainer
                order={this.state.currentOrder}
                orderId={this.state.currentOrderId}
                MealRow={MealRow}
                removeFromCart={this.removeMealFromCart}
                clearCart={this.clearOrder}
                closeCart={this.closeCartModal}
                notify={this.notify}
              /> :
              <div>
                <h3>Order Cart</h3>
                <p>
                  No orders here. Select a meal and click the Add to Cart button
                </p>

              </div>
                }
            </aside>
          </ReactModal>

          <ReactModal
            isOpen={this.state.showMealDetailModal}
            contentLabel="Input Modal"
            className="modal-content"
            onRequestClose={this.closeMealDetailModal}
            shouldCloseOnOverlayClick
            style={{ content: { width: '60%' } }}
          >
            <MealsTable
              meals={this.props.orderMeals}
              total={this.props.total}
              mealsPagination={this.props.mealsPagination}
              loading={this.props.loadingMeals}
              onFetchData={this.onFetchMealData}
              closeMealDetailModal={this.closeMealDetailModal}
              getTrProps={() => ({ style: { cursor: 'pointer' } })}
            />
          </ReactModal>
        </div>

      </div>
    );
  }
}

Order.defaultProps = {
  menu: [],
  orders: [],
  pendingOrders: [],
  orderMeals: [],
  orderConnecting: false,
  loadingMeals: false,
  total: 0,
  pagination: {
    pages: 1,
    limit: 10,
    offset: 0
  },
  mealsPagination: {
    pages: 1,
    limit: 5,
    offset: 0
  },
};

Order.propTypes = {
  dispatch: PropTypes.func.isRequired,
  orders: PropTypes.arrayOf(PropTypes.object),
  menu: PropTypes.arrayOf(PropTypes.object),
  orderMeals: PropTypes.arrayOf(PropTypes.object),
  pagination: PropTypes.shape({
    pages: PropTypes.number,
    limit: PropTypes.number,
    offset: PropTypes.number
  }),
  mealsPagination: PropTypes.shape({
    pages: PropTypes.number,
    limit: PropTypes.number,
    offset: PropTypes.number
  }),
  orderConnecting: PropTypes.bool,
  loadingMeals: PropTypes.bool,
  pendingOrders: PropTypes.arrayOf(PropTypes.object),
  total: PropTypes.number,
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
    firstName: PropTypes.string,
    id: PropTypes.string
  }).isRequired,
};

const mapStateToProps = state => ({
  orderError: state.orderReducer.orderError,
  orderMealsError: state.orderReducer.orderMealsError,
  loadingMeals: state.orderReducer.loadingMeals,
  total: state.orderReducer.total,
  pagination: state.orderReducer.pagination,
  mealsPagination: state.orderReducer.mealsPagination,
  orderConnecting: state.orderReducer.connecting,
  menuConnecting: state.menuReducer.connecting,
  menu: state.menuReducer.menu,
  orders: state.orderReducer.orders,
  pendingOrders: state.orderReducer.pendingOrders,
  orderMeals: state.orderReducer.orderMeals,
  user: state.loginReducer.user.data
});

export { Order };

export default connect(mapStateToProps)(Order);
