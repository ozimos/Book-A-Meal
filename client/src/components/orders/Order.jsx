import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import ReactModal from 'react-modal';

import {
  Accordion,
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';
import SearchInput, { createFilter } from 'react-search-input';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import MealCard3 from '../mealCard/MealCard3';
import MealRow from '../orderCart/MealRow';
import CartContainer from '../orderCart/CartContainer';
import OrderItem from '../orderCart/OrderItem';
import MealCardContainer from '../mealCard/MealCardContainer';
import OrderContainer from '../mealCard/OrderContainer';
import Greeting from '../greeting/Greeting';
import { menuActions, orderActions } from '../../redux/actions';
import ConnectedNav from '../nav/Nav';
import '../../../public/styles/bookameal.scss';
import '../../../public/styles/search-input.css';
import '../../../public/styles/accordion.css';

ReactModal.setAppElement(document.getElementById('root'));
class Order extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      currentOrder: [],
      currentOrderId: '',
      showModal: false,
    };
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
    this.removeFromOrder = this.removeFromOrder.bind(this);
    this.clearOrder = this.clearOrder.bind(this);
    this.addOrder = this.addOrder.bind(this);
    this.addToOrder = this.addToOrder.bind(this);
    this.notify = this.notify.bind(this);
  }
  componentDidMount() {
    this.props.dispatch(menuActions.getMenu());
    this.props.dispatch(orderActions.getUserOrdersByDate());
  }
  handleOpenModal() {
    return this.setState({ showModal: true });
  }
  handleCloseModal() { return this.setState({ showModal: false }); }
  notify = message => toast(message, { className: 'toaster' });
  addToOrder(meal) {
    if (this.state.currentOrderId) {
      return toast('New meals cannot be added to existing orders', { className: 'toaster' });
    }
    const inOrder = this.state.currentOrder.some(elem => elem.id === meal.id);
    if (!inOrder) {
      this.setState(prevState =>
        ({ currentOrder: [...prevState.currentOrder, meal] }));
      toast('Meal has been added to cart', { className: 'toaster' });
    } else {
      toast('Meal is already in cart', { className: 'toaster' });
    }
  }
  addOrder(id) {
    const order = this.props.orders.find(elem => elem.id === id);
    this.setState({ currentOrder: order.Meals, currentOrderId: id });
    toast('Meal has been added to cart for editing', { className: 'toaster' });
  }
  removeFromOrder(id) {
    if (this.state.currentOrder.length <= 1) {
      toast.error('There must be at least one meal in the cart. Use the clear cart button to clear cart', { className: 'toaster' });
    } else {
      this.setState(prevState =>
        ({ currentOrder: prevState.currentOrder.filter(elem => elem.id !== id) }));
    }
  }
  clearOrder() {
    this.setState({ currentOrder: [], currentOrderId: '' });
    this.handleCloseModal();
  }
  searchUpdated = (term) => {
    this.setState({ searchTerm: term });
  }
  postOrder = () => {
    const orderIdArray = this.state.currentOrder.map(meal => meal.id);
    this.props.dispatch(orderActions.postOrder({ currentOrder: orderIdArray }));
  }

  render() {
    const KEYS_TO_FILTERS = ['id', 'title', 'description', 'price'];


    const isMenuSet = this.props.menu.length !== 0;
    const isMealSelected = this.state.currentOrder.length !== 0;
    const isTodayOrder = this.props.orders.length !== 0;
    let filteredMeals;
    if (isMenuSet) {
      filteredMeals = this.props.menu
        .filter(createFilter(this.state.searchTerm, KEYS_TO_FILTERS));
    }
    const { isCaterer, firstName } = this.props.user;
    return (
      <div className="contain">
        <header className="header">
          <ConnectedNav />
        </header>
        <Greeting isCaterer={isCaterer} firstName={firstName} />
        <div className="row">
          <main className="col s12">
            <ToastContainer autoClose={2000} />
            <Accordion accordion={false}>
              <AccordionItem expanded>
                <AccordionItemTitle>
                  <div className="title-element flexbox">
                    <h4 className="long_string">
                  Today&#39;s Menu
                    </h4>
                    <div className="accordion__arrow u-position-relative" role="presentation" />
                  </div>
                </AccordionItemTitle>
                <AccordionItemBody>
                  <div className="flexbox">
                    <SearchInput className="search-input input-field" onChange={this.searchUpdated} />
                    <button className="btn title-button" onClick={this.handleOpenModal} disabled={!isMealSelected} >
                      <p className="cart-notification">Cart<span className="badge">{this.state.currentOrder.length}</span></p>
                    </button>
                  </div>
                  {isMenuSet ? <MealCardContainer
                    meals={filteredMeals}
                    MealCard={MealCard3}
                    addToOrder={this.addToOrder}
                    addClass="scroll"
                  /> :
                  <div>
                  The menu for the day has not been set
                  </div>
                }
                </AccordionItemBody>
              </AccordionItem>
              <AccordionItem>
                <AccordionItemTitle>
                  <div className="title-element flexbox">
                    <h4 className="long_string">
                      Your Orders for Today
                    </h4>
                    <div className="accordion__arrow u-position-relative" role="presentation" />
                  </div>
                </AccordionItemTitle>
                <AccordionItemBody>
                  <p className="mr-auto">
                    {`Orders can only be edited up to ${process.env.ORDER_INTERVAL_HOUR || 4} hours after booking`}
                  </p>
                  { isTodayOrder ? <OrderContainer
                    orders={this.props.orders}
                    OrderItem={OrderItem}
                    addOrder={this.addOrder}
                  /> :
                  <div>
                  You have not placed an order today
                  </div>
                }
                </AccordionItemBody>
              </AccordionItem>
            </Accordion>
          </main>
          <ReactModal
            isOpen={this.state.showModal}
            contentLabel="Input Modal"
            className="modal-content"
            onRequestClose={this.handleCloseModal}
            shouldCloseOnOverlayClick
          >
            <aside className="col s12" >
              {isMealSelected ? <CartContainer
                order={this.state.currentOrder}
                orderId={this.state.currentOrderId}
                MealRow={MealRow}
                removeFromCart={this.removeFromOrder}
                clearCart={this.clearOrder}
                closeCart={this.handleCloseModal}
                notify={this.notify}
              /> :
              <div>
                <h3>Order Cart</h3>
                <p>No orders here. Select a meal and click the Add to Cart button</p>

              </div>
                }
            </aside>
          </ReactModal>
        </div>

      </div>
    );
  }
}
Order.defaultProps = {
  menu: [],
  orders: []
};
Order.propTypes = {
  dispatch: PropTypes.func.isRequired,
  orders: PropTypes.arrayOf(PropTypes.object),
  menu: PropTypes.arrayOf(PropTypes.object),
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
    firstName: PropTypes.string,
    id: PropTypes.string
  }).isRequired,
};
const mapStateToProps = state => ({
  orderError: state.orderReducer.orderError,
  connecting: state.orderReducer.connecting,
  menu: state.menuReducer.menu.Meals,
  orders: state.orderReducer.orders,
  user: state.loginReducer.user.data
});

export { Order };
export default connect(mapStateToProps)(hot(module)(Order));
