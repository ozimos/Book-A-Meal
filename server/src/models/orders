const orders = [{
  id: 1,
  userId: 1,
  menuId: 2,
  mealId: 2,
},
{
  id: 2,
  userId: 1,
  menuId: 1,
  mealId: 1,
}];

export default orders;