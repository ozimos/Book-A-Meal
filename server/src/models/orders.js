export default (sequelize, DataTypes) => {
  const Order = sequelize.define(
    'Order', {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
      }
    },
    {
      scopes: {
        forNonCaterers() {
          return {
            include: [{
              association: 'Meals',
              required: true,
              paranoid: false,
              attributes: ['id', 'userId', 'title', 'description', 'price'],
              through: {
                attributes: ['quantity']
              }
            },
            {
              association: 'User',
              attributes: ['firstName', 'lastName', 'email']
            }]
          };
        },
        forCaterers(userId) {
          return {
            include: [{
              association: 'Meals',
              where: { userId },
              required: true,
              paranoid: false,
              attributes: ['id', 'userId', 'title', 'description', 'price'],
              through: {
                attributes: ['quantity']
              }
            },
            {
              association: 'User',
              attributes: ['firstName', 'lastName', 'email']
            }]
          };
        }
      }
    }
  );

  // Relations
  Order.associate = (models) => {
    Order.belongsTo(models.User, {
      foreignKey: 'userId',
    });
    Order.belongsToMany(models.Meal, {
      through: 'MealOrders',
      foreignKey: 'orderId',
      as: 'Meals',
    });
  };

  return Order;
};
