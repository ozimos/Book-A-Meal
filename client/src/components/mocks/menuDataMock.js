import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};