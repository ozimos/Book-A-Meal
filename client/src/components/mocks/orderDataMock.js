import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};