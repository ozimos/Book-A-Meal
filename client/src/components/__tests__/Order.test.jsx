/* global React:false, shallow:false toJson:false */

import { Order } from '../orders/ConnectedOrder';
import { allOrders } from '../mocks/orderDataMock';
import { allMeals } from '../mocks/mealDataMock';

describe('Order Component', () => {
  const user = { isCaterer: true, firstName: 'user' };

  it('should render without throwing an error', () => {
    expect(shallow(<Order
      dispatch={jest.fn()}
      user={user}
      menu={allMeals.data}
      orders={allOrders.data}
    />)
      .exists(<main className="col-12 col-md-8" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<Order
      dispatch={jest.fn()}
      user={user}
      menu={allMeals.data}
      orders={allOrders.data}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

