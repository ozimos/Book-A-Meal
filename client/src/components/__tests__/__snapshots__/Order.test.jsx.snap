// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Component renders correctly 1`] = `
<div
  className="contain"
>
  <header
    className="header"
  >
    <Connect(Nav) />
  </header>
  <Greeting
    firstName="user"
    isCaterer={true}
  />
  <div
    className="row"
  >
    <main
      className="col s12"
    >
      <ToastContainer
        autoClose={2000}
        bodyClassName={null}
        className={null}
        closeButton={
          <CloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        draggable={true}
        draggablePercent={80}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        pauseOnVisibilityChange={true}
        position="top-right"
        progressClassName={null}
        rtl={false}
        style={null}
        toastClassName={null}
        transition={[Function]}
      />
      <AccordionWrapper
        accordion={false}
        className="accordion"
        onChange={[Function]}
      >
        <AccordionItemWrapper
          accordionStore={
            AccordionContainer {
              "_listeners": Array [],
              "state": Object {
                "accordion": true,
                "items": Array [],
                "onChange": [Function],
              },
            }
          }
          className="accordion__item"
          disabled={false}
          expanded={true}
          hideBodyClassName=""
        >
          <AccordionItemTitleWrapper
            className="accordion__title"
            hideBodyClassName=""
          >
            <div
              className="title-element flexbox"
            >
              <h4
                className="long_string"
              >
                Today's Menu
              </h4>
              <div
                className="accordion__arrow u-position-relative"
                role="presentation"
              />
            </div>
          </AccordionItemTitleWrapper>
          <AccordionItemBodyWrapper
            className="accordion__body"
            hideBodyClassName="accordion__body--hidden"
          >
            <div
              className="flexbox"
            >
              <Search
                caseSensitive={false}
                className="search-input input-field"
                fuzzy={false}
                onChange={[Function]}
                throttle={200}
              />
              <button
                className="btn title-button"
                disabled={true}
                onClick={[Function]}
              >
                <p
                  className="cart-notification"
                >
                  Cart
                  <span
                    className="badge"
                  >
                    0
                  </span>
                </p>
              </button>
            </div>
            <MealCardContainer
              MealCard={[Function]}
              addClass="scroll"
              addToOrder={[Function]}
              meals={
                Array [
                  Object {
                    "id": "abc",
                    "price": 1200,
                  },
                  Object {
                    "id": "def",
                    "price": 2000,
                  },
                ]
              }
            />
          </AccordionItemBodyWrapper>
        </AccordionItemWrapper>
        <AccordionItemWrapper
          accordionStore={
            AccordionContainer {
              "_listeners": Array [],
              "state": Object {
                "accordion": true,
                "items": Array [],
                "onChange": [Function],
              },
            }
          }
          className="accordion__item"
          disabled={false}
          expanded={false}
          hideBodyClassName=""
        >
          <AccordionItemTitleWrapper
            className="accordion__title"
            hideBodyClassName=""
          >
            <div
              className="title-element flexbox"
            >
              <h4
                className="long_string"
              >
                Your Orders for Today
              </h4>
              <div
                className="accordion__arrow u-position-relative"
                role="presentation"
              />
            </div>
          </AccordionItemTitleWrapper>
          <AccordionItemBodyWrapper
            className="accordion__body"
            hideBodyClassName="accordion__body--hidden"
          >
            <p
              className="mr-auto"
            >
              Orders can only be edited up to 
                    15
                     minutes after order is placed
            </p>
            <OrderContainer
              OrderItem={[Function]}
              addOrder={[Function]}
              orders={
                Array [
                  Object {
                    "Meals": Array [
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "abc",
                        "price": 1200,
                      },
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "def",
                        "price": 2000,
                      },
                    ],
                    "User": Object {
                      "email": "asdf",
                      "firstName": "first",
                      "lastName": "last",
                    },
                    "id": "order1",
                    "mealList": Array [
                      "abc",
                      "def",
                    ],
                    "quantityList": Array [
                      2,
                      2,
                    ],
                    "userId": "userId",
                  },
                  Object {
                    "Meals": Array [
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "abc",
                        "price": 1200,
                      },
                      Object {
                        "MealOrders": Object {
                          "quantity": 2,
                        },
                        "id": "def",
                        "price": 2000,
                      },
                    ],
                    "User": Object {
                      "email": "asdf",
                      "firstName": "first",
                      "lastName": "last",
                    },
                    "id": "order2",
                    "mealList": Array [
                      "abc",
                      "def",
                    ],
                    "quantityList": Array [
                      2,
                      2,
                    ],
                    "userId": "userId",
                  },
                ]
              }
            />
          </AccordionItemBodyWrapper>
        </AccordionItemWrapper>
      </AccordionWrapper>
    </main>
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal-content"
      closeTimeoutMS={0}
      contentLabel="Input Modal"
      isOpen={false}
      onRequestClose={[Function]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <aside
        className="col s12"
      >
        <div>
          <h3>
            Order Cart
          </h3>
          <p>
            No orders here. Select a meal and click the Add to Cart button
          </p>
        </div>
      </aside>
    </Modal>
  </div>
</div>
`;
